src/libraries/Microsoft.Extensions.Options/src/OptionsFactory.cs

```diff
@@ -48,7 +48,7 @@ public OptionsFactory(IEnumerable<IConfigureOptions<TOptions>> setups, IEnumerab
         /// <summary>
         /// Returns a configured <typeparamref name="TOptions"/> instance with the given <paramref name="name"/>.
         /// </summary>
-        public TOptions Create(string name)
+        public TOptions Create(string? name)
```

Comment:
Right I think null name is normalized to Options.DefaultName generally, I think null name is only used to signal configuring all option instances in some places if I remember correctly

Reply:
I believe this one is still valid. What would "Create `all`" mean?

If you want to create an Options, you have to pass in a non-null string. And if the Options doesn't have a name, you use `Options.DefaultName` (which is `string.Empty`).

Reply:
You can call nullable APIs from a method that takes in a non-nullable value. I don't see that as a reason the decide what `Create` should be allowed to take.

`null` names means "configure all named options". But the `OptionsFactory.Create` API isn't dealing with "all" options. It is dealing with _this one option_ - the thing it is creating. And saying "to create an option, you need to give it a name or use `Options.DefaultName`" makes sense to me.

Reply:
Yeah, but it's strange that _all_ methods that are called from `Create` accept `null` (not counting `OptionsValidationException` since there's no business logic in there), and there's no much in logic in `Create` itself outside calling those methods. I guess, by the logic you described `CreateInstance` also shouldn't accept `null`.

Marked both `Create` and `CreateInstance`'es `name` parameter as non-nullable.

Reply:
>  by the logic you described CreateInstance also shouldn't accept null.

Correct. That was my thinking as well.